torchft/checkpointing.py

```diff
@@ -117,7 +141,9 @@ def do_GET(self):
                     # validate socket timeout is actually set
                     assert self.connection.gettimeout() == self.timeout
 
```

Comment:
Would it make sense to add an assert to check self._step == -1 (or maybe initialize step to None and check that) to fail quickly rather than to wait and timeout?

Reply:
We actually are doing that which is the issue. Down below we check if self._step == the requested step

We end up with a race condition between send_checkpoint and the remote worker requesting the checkpoint which causes the requester to get an error due to mismatched step

This lock allows us to have some tolerance when requesting a checkpoint. We could busy loop/retry on caller side but the lock makes the code simpler and a bit more efficient